az-cvm-vtpm/src/vtpm/verify.rs

```diff
@@ -53,29 +61,78 @@ impl Quote {
         }
         Ok(())
     }
+
+    /// Verify a Quote's PCR values
+    ///
+    pub fn verify_pcrs(&self) -> Result<(), VerifyError> {
+        let attest = Attest::unmarshall(&self.message)?;
+        let AttestInfo::Quote { info } = attest.attested() else {
+            return Err(VerifyError::Quote(QuoteError::NotAQuote));
+        };
+
+        let pcr_digest = info.pcr_digest();
+
+        // Read hashes of all the PCRs.
+        let mut hasher = Sha256::new();
+        for pcr in self.pcrs.iter() {
+            hasher.update(pcr);
+        }
+
+        let digest = hasher.finalize();
+        if digest[..] != pcr_digest[..] {
+            return Err(VerifyError::PcrMismatch);
+        }
+
+        Ok(())
+    }
 }
 
 #[cfg(test)]
 mod tests {
     use super::*;
 
+    // // Use this code to generate the scriptures for the test on an AMD CVM.
+    //
+    // use az_snp_vtpm::vtpm;
+    // use bincode;
+    // use rsa;
+    // use rsa::pkcs8::EncodePublicKey;
+    // use std::error::Error;
+    // use std::fs;
+    //
+    // fn main() -> Result<(), Box<dyn Error>> {
+    //     // Extract the AK public key.
+    //     let foo = vtpm::get_ak_pub()?.to_public_key_pem(rsa::pkcs8::LineEnding::LF)?;
+    //     fs::write("/tmp/akpub.pem", foo)?;
+    //
+    //     // Save the PCRs into binary file.
+    //     let nonce = "challenge".as_bytes().to_vec();
+    //     let quote = vtpm::get_quote(&nonce)?;
+    //     let quote_encoded: Vec<u8> = bincode::serialize(&quote).unwrap();
+    //     fs::write("/tmp/quote.bin", quote_encoded)?;
+    //
+    //     Ok(())
+    // }
+
     #[cfg(feature = "verifier")]
     #[test]
     fn test_quote_validation() {
         // Can be retrieved by `get_ak_pub()` or via tpm2-tools:
-        // `tpm2_readpublic -c 0x81000003 -f pem -o akpub.pem`
-
+        // sudo tpm2_readpublic -c 0x81000003 -f pem -o akpub.pem
         let pem = include_bytes!("../../test/akpub.pem");
         let pkey = PKey::public_key_from_pem(pem).unwrap();
 
         // Can be retrieved by `get_quote()` or via tpm2-tools:
-        // `tpm2_quote -c 0x81000003 -l sha256:5,8 -q cafe -m quote_msg -s quote_sig`
-        let message = include_bytes!("../../test/quote_msg").to_vec();
-        let signature = include_bytes!("../../test/quote_sig").to_vec();
-        let quote = Quote { signature, message };
+        // For message and signature:
+        // sudo tpm2_quote -c 0x81000003 -l sha256:5,8 -q challenge -m quote_msg -s quote_sig
+        //
+        // For PCR values:
+        // sudo tpm2_pcrread sha256:0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23
+        let quote_bytes: Vec<u8> = include_bytes!("../../test/quote.bin").to_vec();
+        let quote: Quote = bincode::deserialize(&quote_bytes[..]).unwrap();
```

Comment:
bytes are already a slice, no need for conversion

```suggestion
        let quote_bytes = include_bytes!("../../test/quote.bin");
        let quote: Quote = bincode::deserialize(quote_bytes).unwrap();
```